cmd/aws-cost-exporter/main.go

```diff
@@ -1,44 +1,74 @@
 package main
 
 import (
-	"github.com/st8ed/aws-cost-exporter/pkg/collector"
-	"github.com/st8ed/aws-cost-exporter/pkg/fetcher"
-	"github.com/st8ed/aws-cost-exporter/pkg/processor"
-	"github.com/st8ed/aws-cost-exporter/pkg/state"
-
 	"context"
+	"net/http"
+	"os"
 	"os/user"
+	"sync/atomic"
+	"time"
 
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/collectors"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	dto "github.com/prometheus/client_model/go"
-	"github.com/prometheus/common/version"
-
-	"github.com/go-kit/log"
-	"github.com/go-kit/log/level"
 	"github.com/prometheus/common/promlog"
 	"github.com/prometheus/common/promlog/flag"
-
-	"net/http"
-	"os"
-
+	"github.com/prometheus/common/version"
 	"github.com/prometheus/exporter-toolkit/web"
-
+	"golang.org/x/sync/errgroup"
```

Comment:
this is a new dependency added to properly manage the lifecycle of the two concurrent processes:
1. the HTTP server; and
2. the `intervalGatherer`

at the end of the `main` func we call `g.Wait()`, which waits for the goroutines to finish and returns the first error that was found (if any).